MetaMask/metamask-mobile
# fix: Browser dialog prompts in app

## **Description**

Added a function that checks to see if the Webview Tab is currently in focus. If the user navigates away from the webview at any point, the app will stop any JS execution in the webview. If the user remains in the webview tab, JS execution will continue as expected.

## **Related issues**

Fixes: [Browser Dialog prompts can overlay outside of tab](https://github.com/orgs/MetaMask/projects/60/views/16?pane=issue&itemId=43794381)

## **Manual testing steps**

1. Download the following file and visit it in the MM Mobile browser: https://github.com/MetaMask/temp-poc-templates-repo/blob/obs3rv3r-pocs/metamask-poc.html
2. Upon loading the page, switch to another part of the app, such as the settings page or the home page.
3. See that the browser UI prompts still show up over the application
4. Do steps 1-2 again but this time visit another tab in the browser.
5. See that the prompts show up over the other tab

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/95f7aee7-1e70-4a1f-8c43-55a95d3dccc8

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
